EIPS/eip-4824.md

```diff
@@ -0,0 +1,220 @@
+---
+eip: 4824
+title: DAO API
+description: An API standard for decentralized autonomous organizations (DAOs).
+author: Joshua Tan (@thelastjosh), Isaac Patka (@ipatka), Ido Gershtein <ido@daostack.io>, Eyal Eithcowich <eyal@deepdao.io>, Michael Zargham (@mzargham), Sam Furter (@nivida)
+discussions-to: https://ethereum-magicians.org/t/eip-4824-decentralized-autonomous-organizations/8362
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-17
+---
+
+## Abstract
+
+An API standard for decentralized autonomous organizations (DAOs), focused on relating on-chain and off-chain representations of membership and proposals.
+
+## Motivation
+
+DAOs, since being invoked in the Ethereum whitepaper, have been vaguely defined. This has led to a wide range of patterns but little standardization or interoperability between the frameworks and tools that have emerged. Standardization and interoperability are necessary to support a variety of use-cases. In particular, a standard daoURI, similar to tokenURI in [ERC-721](./eip-721), will enhance DAO discoverability, legibility, proposal simulation, and interoperability between tools. More consistent data across the ecosystem is also a prerequisite for future DAO standards.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every EIP-4824 compliant contract MUST implement the EIP4824 interface below:
+
+```solidity
+pragma solidity ^0.4.20;
+
+/// @title EIP-4824 DAOs
+/// @dev See https://eips.ethereum.org/EIPS/eip-4824
+
+interface EIP4824 {
+    /// @notice A distinct Uniform Resource Identifier (URI) pointing to a JSON object following the "EIP-4824 DAO JSON-LD Schema". This JSON file splits into four URIs: membersURI, proposalsURI, activityLogURI, and governanceURI. The membersURI should point to a JSON file that conforms to the "EIP-4824 Members JSON-LD Schema". The proposalsURI should point to a JSON file that conforms to the "EIP-4824 Proposals JSON-LD Schema". The activityLogURI should point to a JSON file that conforms to the "EIP-4824 Activity Log JSON-LD Schema". The governanceURI should point to a flatfile, normatively a .md file. Each of the JSON files named above can be statically-hosted or dynamically-generated.
+    function daoURI() external view returns (string _daoURI);
+}
+```
+
+The EIP-4824 DAO JSON-LD Schema mentioned above.
+
+```json
+{
+	"@context": "http://www.daostar.org/schemas",
+	"type": "DAO",
+	"name": "<name of the DAO>",
+	"description": "<description>",
+	"membersURI": "<URI>",
+	"proposalsURI": "<URI>",
+	"activityLogURI": "<URI>",
+	"governanceURI": "<URI>"
+}
+```
```

Comment:
ERC-721 uses [JSON Schema](https://json-schema.org/) to define its metadata structure. Would you be opposed to using that here?

Reply:
I wouldn't mind either JSON schema or JSON-LD. I'm mostly just concerned with consistent and usable standards. Either way, the validation rules probably should be in this EIP, so that an implementer doesn't need any external resources.